index.js

```diff
@@ -543,10 +543,15 @@ Primus.readable('library', function compile(nodejs) {
   // `ping` interval of the client to ensure that we've sent the server
   // a message before the timeout gets triggered and we get disconnected.
   //
-  if ('number' === typeof this.timeout) {
-    var timeout = this.timeout - 10000;
-    log('adding a custom timeout to the client');
-    client = client.replace('options.ping : 25e3;', 'options.ping : '+ timeout +';');
+  if (this.timeout) {
```

Comment:
If I remember correctly the `typeof` check was needed to allow a value of `0`, but setting the heartbeat timeout to `0` doesn't have much sense as the connection will be killed immediately.
This change assumes that the user want to disable the timeout when setting it to `0`.